File: lib/src/main/java/com/sd/lib/dldmgr/DownloadState.java
package com.sd.lib.dldmgr;

public enum DownloadState
{
    Prepare,
    Downloading,
    Success,
    Error;

    /**
     * 是否处于活动状态，{@link #Prepare}或者{@link #Downloading}
     *
     * @return
     */
    public boolean isActive()
    {
        return this == Prepare || this == Downloading;
    }
}

File: lib/src/main/java/com/sd/lib/dldmgr/DownloadState.kt
package com.sd.lib.dldmgr

enum class DownloadState {
    /** 初始状态 */
    Initialized,

    /** 下载中 */
    Downloading,

    /** 下载成功 */
    Success,

    /** 下载失败 */
    Error;

    /** 是否处于完成状态，[Success]或者[Error] */
    val isFinished: Boolean
        get() = this == Success || this == Error
}